geopandas/tests/test_clip.py

```diff
@@ -0,0 +1,232 @@
+    )
+    return gdf
+
+
+@pytest.fixture
+def single_rectangle_gdf():
+    """Create a single rectangle for clipping. """
+    poly_inters = Polygon([(0, 0), (0, 10), (10, 10), (10, 0), (0, 0)])
+    gdf = gpd.GeoDataFrame([1], geometry=[poly_inters], crs={"init": "epsg:4326"})
+    gdf["attr2"] = "site-boundary"
+    return gdf
+
+
+@pytest.fixture
+def larger_single_rectangle_gdf():
+    """Create a slightly larger rectangle for clipping.
+
+     The smaller single rectangle is used to test the edge case where slivers
+     are returned when you clip polygons. This fixture is larger which
+     eliminates the slivers in the clip return."""
+    poly_inters = Polygon([(-5, -5), (-5, 15), (15, 15), (15, -5), (-5, -5)])
+    gdf = gpd.GeoDataFrame([1], geometry=[poly_inters], crs={"init": "epsg:4326"})
+    gdf["attr2"] = ["study area"]
+    return gdf
+
+
+@pytest.fixture
+def buffered_locations(point_gdf):
+    """Buffer points to create a multi-polygon. """
+    buffered_locs = point_gdf
+    buffered_locs["geometry"] = buffered_locs.buffer(4)
+    buffered_locs["type"] = "plot"
+    return buffered_locs
+
+
+@pytest.fixture
+def donut_geometry(buffered_locations, single_rectangle_gdf):
+    """ Make a geometry with a hole in the middle (a donut). """
+    donut = gpd.overlay(
+        buffered_locations, single_rectangle_gdf, how="symmetric_difference"
+    )
+    return donut
+
+
+@pytest.fixture
+def two_line_gdf():
+    """ Create Line Objects For Testing """
+    linea = LineString([(1, 1), (2, 2), (3, 2), (5, 3)])
+    lineb = LineString([(3, 4), (5, 7), (12, 2), (10, 5), (9, 7.5)])
+    gdf = gpd.GeoDataFrame([1, 2], geometry=[linea, lineb], crs={"init": "epsg:4326"})
+    return gdf
+
+
+@pytest.fixture
+def multi_poly_gdf(donut_geometry):
+    """ Create a multi-polygon GeoDataFrame. """
+    multi_poly = donut_geometry.unary_union
+    out_df = gpd.GeoDataFrame(
+        geometry=gpd.GeoSeries(multi_poly), crs={"init": "epsg:4326"}
+    )
+    out_df = out_df.rename(columns={0: "geometry"}).set_geometry("geometry")
+    out_df["attr"] = ["pool"]
+    return out_df
+
+
+@pytest.fixture
+def multi_line(two_line_gdf):
+    """ Create a multi-line GeoDataFrame.
+
+    This has one multi line and another regular line.
+    """
+    # Create a single and multi line object
+    multiline_feat = two_line_gdf.unary_union
+    linec = LineString([(2, 1), (3, 1), (4, 1), (5, 2)])
+    out_df = gpd.GeoDataFrame(
+        geometry=gpd.GeoSeries([multiline_feat, linec]), crs={"init": "epsg:4326"}
+    )
+    out_df = out_df.rename(columns={0: "geometry"}).set_geometry("geometry")
+    out_df["attr"] = ["road", "stream"]
+    return out_df
+
+
+@pytest.fixture
+def multi_point(point_gdf):
+    """ Create a multi-point GeoDataFrame. """
+    multi_point = point_gdf.unary_union
+    out_df = gpd.GeoDataFrame(
+        gpd.GeoSeries([multi_point, Point(2, 5), Point(-11, -14), Point(-10, -12)]),
```

Comment:
```suggestion
        geometry=gpd.GeoSeries([multi_point, Point(2, 5), Point(-11, -14), Point(-10, -12)]),
```

that should avoid the "rename" line below